rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/metrics/HttpClientMetricsService.java

```diff
@@ -69,17 +69,20 @@ public void update(String uri, String statusDescription, long durationMs) {
      * @param cause Throwable to unwrap
      * @return string description
      */
-
     public static String unwrapExceptionString(Throwable cause) {
-        // Unwrap the failure and return message
-        if (cause instanceof Http.Failure) {
+        // HttpStatusException is a sub-type of Http.Failure: check it first.
+        if (cause instanceof Http.HttpStatusException) {
+            return String.valueOf(((Http.HttpStatusException)cause).getCode());
+        } else if (cause instanceof Http.Failure) {
             final Throwable rootCause = cause.getCause();
-            if (rootCause instanceof EOFException) {
-                return rootCause.getClass().getName();
+            if (rootCause != null) {
+                if (rootCause instanceof EOFException) {
+                    // message is unique for each exception, group them by name
+                    return rootCause.getClass().getName();
+                }
+                return rootCause.toString();
             }
-            return rootCause.toString();
-        } else if (cause instanceof Http.HttpStatusException) {
-            return String.valueOf(((Http.HttpStatusException)cause).getCode());
+            return cause.toString();
         }
         return "exception";
```

Comment:
Not sure 'return "exception"' is very useful. 
Maybe String.valueOf(cause)?

Reply:
Some of the exceptions have string representations that are unique for each instance. I could try `cause.toString()` and monitor whether the cardinality of the status field explodes.